code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -87,6 +87,14 @@
 		state["area"] = "???"
 	return state
 
+/mob/living/critter/flock/hand_attack(atom/target, params)
+	var/datum/handHolder/HH = get_active_hand()
+	if (HH.can_attack && ismob(target) && src.a_intent == INTENT_GRAB)
```

Comment:
What about just checking explicitly for if limb is nanite spray or stunner since those are the ones interested in?

Reply:
Does need some sort of check to avoid calling grab twice for the grip tool. Could addon a check for `can_hold_items` being `FALSE` which would be a lot faster the istype checking